### PR TITLE
Correctly handle ignored services in instance states

### PR DESCRIPTION
**Pull request by duncanhall**
This change prevents services marked as `Ignore` from affecting the deployment status of an instance (if a service is ignored, not having it installed should not mark a deployment as failed).

Moreover, the `getInstanceState` module has been refactored to break up a dense area of code that was difficult to test. Tests have been added to cover all combinations of service states.

This also fixes or removes test cases with duplicate names (Team City doesn't know how to identify these separately)

### CODE REVIEW

File: server/test/modules/environment-state/deleteTargetStateTest.js [hunk 1]
```diff
@@ -67,7 +67,7 @@ describe('deleteTargetState', () => {
   });
 
 
-  describe('byService', () => {
+  describe('byServiceVersion', () => {
```
Comment (Merlin-Taylor) [inline server/test/modules/environment-state/deleteTargetStateTest.js]: Well spotted!
